hudi-common/src/main/java/org/apache/hudi/BaseHoodieTableFileIndex.java

```diff
@@ -166,6 +167,11 @@ public Map<String, List<FileSlice>> listFileSlices() {
         .collect(Collectors.toMap(e -> e.getKey().path, Map.Entry::getValue));
   }
 
+  public int getFileSlicesCount() {
+    return cachedAllInputFileSlices.values().stream()
+        .reduce(0, (count, fileSlices) -> count + fileSlices.size(), Integer::sum);
```

Comment:
This is not quite obvious.  Do sth like `cachedAllInputFileSlices.values().stream().mapToInt(List::size).sum()` instead?

Reply:
SG. Will take up as a follow-up

Reply:
Addressed